### ISSUE TITLE
[BUG] False case of detected unused local variable

### ISSUE THREAD

**Issue by dmaicher** [phpmd/phpmd#318]
In the first line phpmd detects an unused local variable which is true but I cannot change the fact. So it should not complain about it? :)

`Avoid unused local variables such as '$req'.`

```php
$this->handler->expects($this->once())->method('handle')->willReturnCallback(function (HotelAvailRQ $req, HotelAvailRS $res) {
    $res->addRoomStay(new RoomStay(new RoomType('foo', 'bar', 'foo'), new RatePlan('foo', 'bar', 'foo')));
});
```

I'm using version PHPMD 2.3.2.

**Comment by ravage84** [comment]
As you already said, it works as advertised.

I guess you are probably best off with suppressing certain warnings at certain places in your code:
http://phpmd.org/documentation/suppress-warnings.html
